tsl/src/continuous_aggs/refresh.c

```diff
@@ -290,15 +291,71 @@ log_refresh_window(int elevel, const ContinuousAgg *cagg, const InternalTimeRang
 		 DatumGetCString(OidFunctionCall1(outfuncid, end_ts)));
 }
 
+#define MAX_INVALIDATIONS_PER_REFRESH 10
+
+/*
+ * Get the limit on number of invalidation-based refreshes we allow per refresh call.
+ *
+ * Use a session variable for now, for debugging and tweaking. Might promote
+ * to full GUC if useful.
+ */
+static long
+get_max_invalidations_per_refresh(void)
+{
+	const char *max_invalidations_setting =
+		GetConfigOption("timescaledb.max_invalidations_per_refresh", true, false);
+	long max_invalidations = MAX_INVALIDATIONS_PER_REFRESH;
+
+	if (max_invalidations_setting)
+	{
+		char *endptr = NULL;
+
+		/* Not using pg_strtol here since we don't want to throw error in case
+		 * of parsing issue */
+		max_invalidations = strtol(max_invalidations_setting, &endptr, 10);
+
+		if (*endptr != '\0')
+			max_invalidations = MAX_INVALIDATIONS_PER_REFRESH;
```

Comment:
What about trailing spaces? Shouldn't that be handled? Also, AFAICT, the empty string will result in 0 max invalidation, which might not want be what you want.

Reply:
I can add handling for trailing spaces, but I don't think this is necessary for something that is not an official GUC. Falling back to the default value or even 0 is OK. All of these will work.

Reply:
Do we need to check `strtol` for overflows as well? errno `ERANGE` or endptr is enought?

Reply:
In my understanding, checking endptr is enough unless we want to understand more about what failed.

Reply:
FWIW, if the integer value is too big it will just return LONG_MAX. Added test.